singularity-eos/eos/eos_base.hpp

```diff
@@ -426,12 +429,13 @@ class EosBase {
   }
 
   // Tooling for modifiers
-  PORTABLE_FORCEINLINE_FUNCTION
-  bool IsModified() const { return false; }
-  PORTABLE_FORCEINLINE_FUNCTION
-  auto UnmodifyOnce() { return *static_cast<CRTP *>(this); }
-  PORTABLE_FORCEINLINE_FUNCTION
-  auto GetUnmodifiedObject() { return *static_cast<CRTP *>(this); }
+  inline constexpr bool IsModified() const { return false; }
```

Comment:
This got rid of some warnings that I couldn't otherwise get rid of @Yurlungur

Reply:
Is the message then that this introspection isn't compatible with being on device?

Reply:
I think this should be compatible (`constexpr` is device callable) but this was the only way I could get rid of the warnings so I went with it.

Reply:
:+1: if it works, it works. I don't see why it can't be constexpr.